File: src/app/components/models/DemandePret.ts

import { User } from './user';
import { StatusDemande } from './StatusDemande';

export class DemandePret {
  idDemandePret?: number;
  user?: User;
  montant?: number;
  tauxInteret?: number;
  statusDemande?: StatusDemande;
  dateDemande?: Date;
  dateMAJ?: Date;

}

File: src/main/java/tn/esprit/spring/Entity/DemandePret.java
package tn.esprit.spring.Entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class DemandePret {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    private Long idDemandePret;
    @ManyToOne
    private User user;
    private double montant;
    private double tauxInteret;
    private StatusDemande statusDemande;
    private Date dateDemande;
    private Date dateMAJ;

}
